transformer_engine/pytorch/module/base.py

```diff
@@ -632,9 +649,9 @@ def _get_fp8_params(self) -> Union[List[torch.Tensor], None]:
     # assume FP8 execution.
     def init_fp8_metadata(self, num_gemms: int = 1) -> None:
         """Initialize fp8 related metadata and tensors during fprop."""
-        self.fp8_parameters = FP8GlobalStateManager.with_fp8_parameters()
-        self.fp8 = FP8GlobalStateManager.is_fp8_enabled()
-        self.fp8_calibration = FP8GlobalStateManager.is_fp8_calibration()
+        self._fast_setattr("fp8_parameters", FP8GlobalStateManager.with_fp8_parameters())
```

Comment:
I wonder if we couldn't instead just do something like
```python
te_params = self.get_te_params()  # calls _fast_getattr internally, te_params is a normal object
te_params.fp8_parameters = FP8GlobalStateManager.with_fp8_parameters()
```

Reply:
Otherwise it will be hard to enforce everybody using only the `_fast_get/setattr` I think.

Reply:
Even better would be storing these attrs in `fp8_meta` or some other dict.

Reply:
I tried wrapping these attrs in a property that internally calls `_fast_setattr`, but that added ~10 us overhead (probably from the extra indirection when getting the attrs). I think it's a good idea to refactor these frequently changed attrs so they are not held directly by the module, but I think that would be beyond the scope of this PR.

Reply:
Moving this logic into the `__setattr__` function makes things a little cleaner. It adds ~2 us overhead, but it's still a win of ~6 us compared to the baseline.